File: examples-salvo/semi-complex/cephyr/06_lis.c
#include <stdio.h>
int
main (void)
{
  int a[] = { 10, 9, 2, 5, 3, 7, 101, 18 }, d[8], best = 0;
  for (int i = 0; i < 8; i++)
    {
      d[i] = 1;
      for (int j = 0; j < i; j++)
        if (a[j] < a[i] && d[j] + 1 > d[i])
          d[i] = d[j] + 1;
      if (d[i] > best)
        best = d[i];
    }
  printf ("%d\n", best);
}

File: examples-salvo/semi-complex/moonix/06_lis.lua
local a = { 10, 9, 2, 5, 3, 7, 101, 18 }
local dp = {}
local best = 0
for i = 1, #a do
	dp[i] = 1
	for j = 1, i - 1 do
		if a[j] < a[i] and dp[j] + 1 > dp[i] then
			dp[i] = dp[j] + 1
		end
	end
	if dp[i] > best then
		best = dp[i]
	end
end
print(best)
